cmd/lintwalk/main.go

```diff
@@ -2,18 +2,43 @@ package lintwalk
 
 import (
 	"flag"
+	"go/build"
+	"go/parser"
+	"go/token"
 	"log"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"regexp"
+	"runtime"
+	"strings"
 )
 
+func packagePath() []string {
+	res := []string{runtime.GOROOT(), build.Default.GOPATH}
+
+	if p, ok := os.LookupEnv("GOPATH"); ok {
+		res = append(res, strings.Split(p, ":")...)
+	}
+	return res
+}
+
+func getPackagePrefix(dir string) string {
+	for _, p := range packagePath() {
+		if filepath.HasPrefix(dir, p) {
+			if res, err := filepath.Rel(filepath.Join(p, "src"), dir); err == nil {
+				return res
+			}
+		}
+	}
+	return ""
+}
+
 // Main implements gocritic sub-command entry point.
 func Main() {
 	enable := flag.String("enable", "all",
 		`forwarded to linter "as is"`)
-	exclude := flag.String("exclude", "testdata/|vendor/|builtin/",
+	exclude := flag.String("exclude", "(.*test.*)|(.*vendor.*)|(.*builtin.*)",
```

Comment:
This will skip more than required.
Older pattern matched `/test/` somewhere in the path, new regexp will match things like `testify` or `contest` (not related to tests at all). Is that intentional?